identity/app/actions/AuthenticatedActions.scala

```diff
@@ -19,43 +21,49 @@ class AuthenticatedActions(
     authService: AuthenticationService,
     identityApiClient: IdApiClient,
     identityUrlBuilder: IdentityUrlBuilder,
-    controllerComponents: ControllerComponents) extends Logging with Results {
+    controllerComponents: ControllerComponents,
+    newsletterService: NewsletterService,
+    idRequestParser: IdRequestParser) extends Logging with Results {
 
   private val anyContentParser: BodyParser[AnyContent] = controllerComponents.parsers.anyContent
   private implicit val ec: ExecutionContext = controllerComponents.executionContext
 
   def redirectWithReturn(request: RequestHeader, path: String): Result = {
     val returnUrl = URLEncoder.encode(identityUrlBuilder.buildUrl(request.uri), "UTF-8")
 
-    val paramsToPass =
-      List("INTCMP", "email")
-        .flatMap(name => request.getQueryString(name).map(value => s"&$name=$value"))
-        .mkString
+    val redirectUrlWithParams = identityUrlBuilder.appendQueryParams(path, List(
```

Comment:
Yes this is correct usage: see [example](https://github.com/guardian/frontend/blob/23966052b872108014bd441dc651ab6e9d91365f/identity/test/services/IdentityUrlBuilderTest.scala#L67)